src/xlsx/functions/worksheet.ts

```diff
@@ -110,7 +110,7 @@ export function addRows(
         let cellNode = escapeXml``;
         // Either formula or static value inside the cell
         if (content?.startsWith("=") && value !== undefined) {
-          const res = addFormula(content, value);
+          const res = addFormula(content, value, sheet.arrayFormulaCells[xc]);
```

Comment:
Actually, I'd remove the `else` in the evaluation plugin and change this like this. I think it would also work
```suggestion
          const res = addFormula(content, value, sheet.arrayFormulaCells[xc] ?? xc);
```